crates/consensus-logic/src/sync_manager.rs

```diff
@@ -133,23 +136,40 @@ pub fn start_sync_tasks<E: ExecEngineCtl + Sync + Send + 'static>(
         )
     });
 
-    // Prepare the client worker state and start the thread for that.
-    let client_worker_state = worker::WorkerState::open(params.clone(), storage.clone())?;
-    let st_ch = status_channel.clone();
-
-    executor.spawn_critical("client_worker_task", move |shutdown| {
-        worker::client_worker_task(shutdown, client_worker_state, btc_block_rx, st_ch)
-    });
-
     Ok(SyncManager {
         params,
         fc_manager_tx: fcm_tx,
-        csm_controller,
         asm_controller,
         status_channel,
     })
 }
 
+fn spawn_csm_listener<W: strata_asm_worker::WorkerContext + Send + Sync + 'static>(
+    executor: &TaskExecutor,
+    params: Arc<Params>,
+    storage: Arc<NodeStorage>,
+    status_channel: StatusChannel,
+    asm_monitor: &strata_service::ServiceMonitor<strata_asm_worker::AsmWorkerService<W>>,
```

Comment:
This is the kind of situation I wanted to avoid that I was gesturing when I mentioned that we should slim down the service monitor's def to just have the monitor type and not be generic over the whole service.  Because now the CSM crate is required to know abou the ASM worker context trait.

Reply:
done